.github/workflows/publish-to-pypi.yml

```diff
@@ -0,0 +1,187 @@
+      - run: |
+          rm -rf airflow_clickhouse_plugin
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v4.4.0
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Install dependencies
+        run: |
+          VERSION=$(echo "${{ github.ref }}" | sed -e 's,.*/\(.*\),\1,')
+          [[ "${{ github.ref }}" == "refs/tags/"* ]] && VERSION=$(echo $VERSION | sed -e 's/^v//')
+          python -m pip install --upgrade pip
+          python -m pip install -r requirements.txt --ignore-installed apache-airflow==${{ matrix.airflow-version }}
+          python -m pip install --force-reinstall --index-url https://test.pypi.org/simple --extra-index-url https://pypi.org/simple airflow-clickhouse-plugin[pandas]==$VERSION
+      - name: Run unit tests
+        run: |
+          python -m unittest discover -s tests/unit
+  integration-tests-test:
+    name: Running integration tests for TestPyPI
+    needs: build-n-publish-test
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: [ '3.7' , '3.8', '3.9', '3.10', '3.11' ]
+        airflow-version: [ '2.0.2' , '2.2.5', '2.3.4', '2.4.3', '2.5.0' ]
+    services:
+      clickhouse:
+        image: yandex/clickhouse-server
+        ports:
+          - 9000/tcp
+    steps:
+      - uses: actions/checkout@v3
+      - run: |
+          rm -rf airflow_clickhouse_plugin
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v4.4.0
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Install dependencies
+        run: |
+          VERSION=$(echo "${{ github.ref }}" | sed -e 's,.*/\(.*\),\1,')
+          [[ "${{ github.ref }}" == "refs/tags/"* ]] && VERSION=$(echo $VERSION | sed -e 's/^v//')
+          python -m pip install --upgrade pip
+          python -m pip install -r requirements.txt --ignore-installed apache-airflow==${{ matrix.airflow-version }}
+          python -m pip install --force-reinstall --index-url https://test.pypi.org/simple --extra-index-url https://pypi.org/simple airflow-clickhouse-plugin[pandas]==$VERSION
+      - name: Run tests on ClickHouse server
+        env:
+          AIRFLOW_CONN_CLICKHOUSE_DEFAULT: "clickhouse://localhost:${{ job.services.clickhouse.ports['9000'] }}"
+        run: |
+          python -m unittest discover -s tests/integration
+
+  build-n-publish-public:
+    name: Publish to PyPI
+    needs: [ unit-tests-test, integration-tests-test ]
+    runs-on: ubuntu-latest
+    steps:
+      - name: Download built package
+        uses: actions/download-artifact@v3
+        with:
+          name: dist
+          path: dist/
+      - name: Publish to PyPI
+        uses: pypa/gh-action-pypi-publish@release/v1
+        with:
+          password: ${{ secrets.PYPI_API_TOKEN }}
+      - name: Wait publishing to PyPI
+        uses: juliangruber/sleep-action@v2.0.0
+        with:
+          time: 300s
```

Comment:
Isn't there anything like a sensor to wait for PyPI new version to appear?

Reply:
I could not find anything similar, so I inserted such a crutch. 
There is something similar, but not the same: https://github.com/Verbalinsurection/track-pypi-version.
I can try writing the script myself using a while loop, checking the available version through poetry (something like this: `poetry search airflow-clickhouse-plugin | grep airflow-clickhouse-plugin | grep -oP '\(\K[^)]+ '`)

Reply:
```
AVAILABLE="poetry search airflow-clickhouse-plugin | grep airflow-clickhouse-plugin | grep -oP '\(\K[^)]+'"
until [ $(eval "$AVAILABLE") == $REQUIRED ]
do
  sleep 2
done
```